scripts/changePartnerShares.ts

```diff
@@ -0,0 +1,45 @@
+import { BigNumberish } from 'ethers';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { ICollector } from 'typechain-types/contracts/Collector';
+import { parseJsonFile } from './utils';
+
+export type ChangePartnerSharesArg = {
+  collectorAddress: string;
+  partnerConfig: string;
+  gasLimit?: BigNumberish;
+};
+
+export type PartnerConfig = {
+  partners: ICollector.RevenuePartnerStruct[];
+};
+
+const defaultTxGas = 150000;
+
+export const changePartnerShares = async (
+  {
+    collectorAddress,
+    partnerConfig,
+    gasLimit = defaultTxGas,
+  }: ChangePartnerSharesArg,
+  hre: HardhatRuntimeEnvironment
+) => {
+  const { ethers } = hre;
+
+  const parsedPartnerConfig = parseJsonFile<PartnerConfig>(
+    partnerConfig
+  ) as PartnerConfig;
+  const collector = await ethers.getContractAt('Collector', collectorAddress);
+
+  try {
+    await collector.updateShares(parsedPartnerConfig.partners, { gasLimit });
```

Comment:
Quick question, does this not need the 'from' parameter specified? I mention this because the contract method restricts that only the contract owner can call it.

Reply:
not in ethers.

Reply:
@ironFe93 That's a good point, I'll create a task to be able to specify an account at the configuration level.